crypto/src/main/java/org/springframework/security/crypto/scrypt/SCryptPasswordEncoder.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright 2002-2012 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.security.crypto.scrypt;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.springframework.security.crypto.codec.Hex;
+import org.springframework.security.crypto.codec.Utf8;
+import org.springframework.security.crypto.keygen.KeyGenerators;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.bouncycastle.crypto.generators.SCrypt;
+
+
+/**
+ * Implementation of PasswordEncoder that uses the SCrypt hashing function. Clients
+ * can optionally supply a cpu cost parameter, a memory cost parameter and a parallelization parameter.
+ * 
+ * @author Shazin Sadakath
+ *
+ */
+public class SCryptPasswordEncoder implements PasswordEncoder {
+    
+    private final Log logger = LogFactory.getLog(getClass());
+    
+    private final int cpuCost;
+    
+    private final int memoryCost;
+    
+    private final int parallelization;
+    
+    private final byte[] saltBytes;
+    
+    public SCryptPasswordEncoder() {
+        this(new String(KeyGenerators.secureRandom().generateKey()), 16384, 8, 1);

```

Comment:
Where did you get your default values? What do they imply (i.e. how much memory is used, how much time does it take on your CPU, etc)?